### ISSUE TITLE
Get peer-to-peer code out of device.hpp

### ISSUE THREAD

**Issue by eyalroz** [eyalroz/cuda-api-wrappers#154]
We currently have some peer-to-peer-related code in `device.hpp` - some type and enum definitions and a couple of functions:

1. None of that code is used inside the `device_t` class itself, right now.
2. The definitions are redundant with those in `peer_to_peer.hpp`
3. The function taking `device::id_t`'s is undesirable in general - see issue #31 .
4. The function taking `device_t`'s is mismatched with its definition in `multi_wrapper_impls.hpp` - which doesn't even require multiple wrappers.

In short, it's a bit of a mess. Let's fix it.
